### ISSUE TITLE
Install Firefox Beta from a link, instead of downloading the .xpi 

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/ipfs-companion#406]
Right now [Firefox Beta Install Link](https://ipfs.io/ipfs/QmR8W5wg8BuAyBTruHnHovfWRavwvidVh3qtyinXi6NnLa)  triggers download instead of install   and users need to perform [this](https://support.mozilla.org/en-US/kb/unable-install-add-ons-extensions-or-themes#w_you-are-asked-to-download-the-add-on-rather-than-installing-it) workaround. According to [this](https://stackoverflow.com/a/12308652) we need to find a way of hosting `.xpi` thet sets mime-type to `application/x-xpinstall`

**Comment by lidel** [comment]
Interestingly, inline installation works fine with github URLs, eg. 
- https://github.com/ipfs-shipyard/ipfs-companion/releases/download/v2.2.0beta2/ipfs_companion_dev_build_ea5c2af-2.2.0.7583-an.fx.xpi

Would be cool to fix it at IPFS gateway level, if not we may consider switching to it in the future.

**Comment by lidel** [comment]
Not sure how useful is this journey, but writing it down in case someone stumbles upon similar problem in future.

### Problem with reliable MIME type detection 

MIME type returned by `go-ipfs` gateways is a tricky business.

Content-based type sniffing relies on [net/http/sniff.go](https://golang.org/src/net/http/sniff.go) which returns `application/zip` for `.xpi` file (because in fact, `.xpi` IS just a ZIP file with different extension :performing_arts:)

Yes,  we have `.xpi` in the filename, but overall, the expected behaviour is unknown:

- MIME sniffing could be potentially dangerous on the gateway https://github.com/ipfs/notes/issues/222
- [RFC] don't sniff the filename to determine the content type https://github.com/ipfs/go-ipfs/pull/4545

In short, `.xpi` loaded from a gateway will probably require download + manual install (as noted [here](https://support.mozilla.org/en-US/kb/unable-install-add-ons-extensions-or-themes#w_you-are-asked-to-download-the-add-on-rather-than-installing-it))

### Solution

**TL;DR** Inline install works fine with Github Release Assets.

After some troubleshooting I can confirm that `.xpi` published at [ipfs-companion/releases](https://github.com/ipfs-shipyard/ipfs-companion/releases) are returned with valid `Content-Type`:

```
curl -L -v 'https://github.com/ipfs-shipyard/ipfs-companion/releases/download/v2.2.0beta3/ipfs_companion_dev_build_e9f77f7-2.2.0.7921-an.fx.xpi' > /dev/null

(...)

< HTTP/1.1 200 OK
< x-amz-id-2: ZRLlhwF97txonNyzOrTidHsSPB26X9fK28ED7jvJ/WucKFewJxpJwZGuMc210xHCzxdoPQRUM7o=
< x-amz-request-id: DC5C78C2B6C04AC2
< Date: Wed, 07 Mar 2018 18:31:07 GMT
< Last-Modified: Tue, 06 Mar 2018 21:32:09 GMT
< ETag: "8e951418f5c81f8e674cc0e772c53d62"
< Content-Disposition: inline; filename=ipfs_companion_dev_build_e9f77f7-2.2.0.7921-an.fx.xpi
< Accept-Ranges: bytes
< Content-Type: application/x-xpinstall
< Content-Length: 3752806
< Server: AmazonS3
```

I switched latest beta channel to use release assets for hosting:

https://github.com/ipfs-shipyard/ipfs-companion/blob/5fad3c64032e77e2d701f7092670f242c8b3cbf5/ci/firefox/update.json#L13
